### PR TITLE
Experimental new debug events API - Draft

### LINKED ISSUE #1349: uncompounding dataset profile's strips metadata
## Description

The uncompound path doesn't copy of the `DatasetProfileView` metadata
 
We need to copy over metadata here when creating a new DatasetProfileView:
https://github.com/whylabs/whylogs/blame/mainline/python/whylogs/migration/uncompound.py#L159

- [x] I have reviewed the [Guidelines for Contributing](https://github.com/whylabs/whylogs/blob/mainline/.github/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/whylabs/whylogs/blob/mainline/.github/CODE_OF_CONDUCT.md).

### PR DESCRIPTION
**Pull request by jamie256**
## Description
Add new log_debug_events, and log_single_image

- [ ] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

File: python/whylogs/extras/image_metric.py [hunk 1]
```diff
@@ -278,5 +281,92 @@ def resolve(self, name: str, why_type: DataType, column_schema: ColumnSchema) ->
...
+        time_tuple = time.gmtime(timestamp)
+
+    feature_name = f"{column_name}_" if column_name else ""
+
+    timed_filename = f"{prefix}{feature_name}{trace_id}.{time.strftime(format, time_tuple)}.json"
+    return timed_filename
+
+
+# put the segment in here, tags
+def log_debug_event(
```
Comment (richard-rogers) [inline python/whylogs/extras/image_metric.py:303]: Should probably be in a different file?

File: python/whylogs/extras/image_metric.py [hunk 2]
```diff
@@ -278,5 +281,92 @@ def resolve(self, name: str, why_type: DataType, column_schema: ColumnSchema) ->
...
+    filename = _get_timed_file_name(trace_id=trace_id, column_name=column_name, timestamp=timestamp)
+    # TODO: integrate with DebugEvents service, but for now save the
+    # debug_event dictionary as a JSON file
+    with open(filename, "w") as json_file:
+        json.dump(debug_event, json_file)
+
+    logger.info(f"debug_event saved to {filename}")
+
+
+def log_single_image(
```
Comment (richard-rogers) [inline python/whylogs/extras/image_metric.py:337]: Should we integrate this with `log_image()` ?

File: python/whylogs/extras/image_metric.py (hunk 1 above)
Comment (jamie256) [reply python/whylogs/extras/image_metric.py:303]: agreed

File: python/whylogs/extras/image_metric.py (hunk 2 above)
Comment (jamie256) [reply python/whylogs/extras/image_metric.py:337]: yeah, we can. I have this separate while we are evaluating the new behavior. It has some quirks that might justify keeping something like this separate.

**Pull request closed without merging**